### DISCUSSION TITLE
Device Authorization Grant via Mobile App without user consent

### DISCUSSION THREAD

**Discussion by eratolekov** [keycloak/keycloak#19858] [Q&A]
Hi there

Is it possible to invoke `{{keycloak_url}}/realms/{{realm}}/device` endpoint with `device_user_code` and end-user's `access token`, and without user's consent?

This case is usable for device authorization via native mobile app for authenticated users.
Mobile app is already contained user's access token. And mobile app is first-party application.

**Comment by jbman** [comment]
The device authorization grant is independent from consent. You can use without a consent screen presented to the user.
Consent setting are configured per client scope, see  https://www.keycloak.org/docs/latest/server_admin/#consent-related-settings

**Reply by eratolekov** [reply]
> Consent setting are configured per client scope, see https://www.keycloak.org/docs/latest/server_admin/#consent-related-settings

In my case _Consent required_ is disabled for client already.

At the same time when user opens the `verification_uri_complete` from `{{keycloak_url}}/realms/{{realm}}/device` response, the consent is shown after login.


But main question, keycloak is capable to provide device authorization flow without web interaction, only rest api ?

My keycloak is 20.0.5

**Comment by sebastienrospars** [comment]
Hello everyone,

I wanted to chime in as I have the same requirement in my mobile application. I am also looking for a way to validate the device_user_code flow without opening a WebView using the `verification_uri_complete`.

Instead, I would like to complete the device authorization flow using the user’s API token to validate the authentication, similar to how apps like Google and Netflix handle their device sign-ins.

I'm trying to create an SPI RealmResourceProvider to create an endpoint to do this but I'm getting stuck.

@eratolekov do you found any solution to do this ? 

Thank you in advance for any updates or potential solutions on this!

**Comment by danifr** [comment]
For completeness it seems intended: "Force consent for device verification flow when there are no client scopes to approve by adding a default client scope to approve" https://github.com/keycloak/keycloak/pull/27474

Original issue: https://github.com/keycloak/keycloak/issues/26100

I also wanted to find a way to disable that consent screen :/
